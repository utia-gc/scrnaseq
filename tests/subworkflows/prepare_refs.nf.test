nextflow_workflow {

    name "Test Workflow Prepare_Refs"
    script "subworkflows/prepare_refs.nf"
    workflow "Prepare_Refs"

    test("Prepare_Refs successfully unzips and emits GZIP-compressed genome and GTF files.") {
        // instantiate helper objects
        def refs = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = file("${projectDir}/${refs.getGenomeGzip()}")
                input[1] = file("${projectDir}/${refs.getGTFGzip()}")
                """
            }
        }

        then {
            // test workflow success
            assert workflow.success
            assert workflow.trace.succeeded().size() == 3

            // test genome decompressed
            assert workflow.out.genome
            with(workflow.out.genome) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa/
            }

            // test genome fasta indexed
            assert workflow.out.genome_index
            with(workflow.out.genome_index) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa.fai/
            }

            // test annotations decompressed
            assert workflow.out.annotations
            with(workflow.out.annotations) {
                assert size() == 1
                assert get(0) ==~ /^.*\/annotations_I.gtf/
            }

            // test reproducibility
            assert snapshot(workflow.out).match()
        }


    }


    test("Prepare_Refs successfully emits uncompressed genome and GTF files when uncompressed inputs are given.") {
        // instantiate helper objects
        def refs = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = file("${projectDir}/${refs.getGenome()}")
                input[1] = file("${projectDir}/${refs.getGTF()}")
                """
            }
        }

        then {
            // test workflow success
            assert workflow.success
            assert workflow.trace.succeeded().size() == 1

            // test genome decompressed
            assert workflow.out.genome
            with(workflow.out.genome) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa/
            }

            // test genome fasta indexed
            assert workflow.out.genome_index
            with(workflow.out.genome_index) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa.fai/
            }

            // test annotations decompressed
            assert workflow.out.annotations
            with(workflow.out.annotations) {
                assert size() == 1
                assert get(0) ==~ /^.*\/annotations_I.gtf/
            }

            // test reproducibility
            assert snapshot(workflow.out).match()
        }

    }

}
