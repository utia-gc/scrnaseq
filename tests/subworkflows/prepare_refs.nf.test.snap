{
    "Prepare_Refs successfully unzips and emits GZIP-compressed fasta and GTF files.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "1": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "annotations": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "fasta": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:38:55+0000"
    },
    "successfully emits uncompressed fasta and GTF files when uncompressed inputs are given.": {
        "content": [
            {
                "0": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ],
                "1": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "annotations": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "fasta": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:38:55+0000"
    },
    "successfully unzips and emits GZIP-compressed fasta and GTF files.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "1": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "annotations": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "fasta": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:38:55+0000"
    },
    "Prepare_Refs successfully unzips and emits GZIP-compressed genome and GTF files.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "1": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "annotations": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "genome": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:38:55+0000"
    },
    "Prepare_Refs successfully emits uncompressed fasta and GTF files when uncompressed inputs are given.": {
        "content": [
            {
                "0": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ],
                "1": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "annotations": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "fasta": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:38:55+0000"
    },
    "Prepare_Refs unzips and outputs GZIP-compressed genome and GTF files.": {
        "content": [
            {
                "0": [
                    "GRCh38.p14_genomic_NC_000021.9.fna:md5,7decb2f4cab1bc058b24c48459469232"
                ],
                "1": [
                    "GRCh38.p14_genomic_NC_000021.9.gtf:md5,5dbd6627231b875a9e62136f8dfe2923"
                ],
                "annotations": [
                    "GRCh38.p14_genomic_NC_000021.9.gtf:md5,5dbd6627231b875a9e62136f8dfe2923"
                ],
                "genome": [
                    "GRCh38.p14_genomic_NC_000021.9.fna:md5,7decb2f4cab1bc058b24c48459469232"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:38:55+0000"
    },
    "Prepare_Refs successfully emits uncompressed genome and GTF files when uncompressed inputs are given.": {
        "content": [
            {
                "0": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ],
                "1": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "annotations": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "genome": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ]
            }
        ],
        "timestamp": "2023-08-22T16:38:55+0000"
    }
}